scripts/azureml-assets/azureml/assets/publish_utils.py

```diff
@@ -523,11 +555,18 @@ def create_asset(asset: AssetConfig, registry_name: str, ml_client: MLClient, ve
                     return False
         elif asset.type == AssetType.MODEL:
             version = asset.version
-            model_config = asset.extra_config_as_object()
+            model_config: ModelConfig = asset.extra_config_as_object()
             if not prepare_model_for_registration(model_config, asset.spec_with_path, Path(temp_dir), ml_client,
                                                   copy_updater):
                 logger.log_error("Failed to prepare model")
                 return False
+        elif asset.type == AssetType.DATA:
+            version = asset.version
+            data_config: DataConfig = asset.extra_config_as_object()
+            if data_config is not None and not prepare_data_for_registration(data_config, asset.spec_with_path,
```

Comment:
No need for the `data_config is not None` check as extra_config_as_object() will raise an exception if it fails or has an unsupported data type, although [DATA is supported](https://github.com/Azure/azureml-assets/blob/7e08c29e06fce8110eb216d79a6a9f4e8ccafe0e/scripts/azureml-assets/azureml/assets/config.py#L1422).

Reply:
I added it because of how local files is currently supported, such as if it's `path: my_local_folder`. It looks like model configs support `type: local`, do you think we should add that functionality in as well? https://github.com/Azure/azureml-assets/blob/7e08c29e06fce8110eb216d79a6a9f4e8ccafe0e/scripts/azureml-assets/azureml/assets/config.py#L738

Reply:
The asset variable above points to an AssetConfig object, which when you call its extra_config_as_object() function will either return one of the supported <type>Config objects or raise an exception. See [the code](https://github.com/Azure/azureml-assets/blob/7e08c29e06fce8110eb216d79a6a9f4e8ccafe0e/scripts/azureml-assets/azureml/assets/config.py#L1403). AssetConfig just contains metadata about the asset, but not the specific path and type properties you're referring to.